addon/validations/factory.js

```diff
@@ -153,68 +159,105 @@ function processDefaultOptions(validations = {}) {
+  const TopLevelProps = createTopLevelPropsMixin(validatableAttributes);
+
+  // Create the `attrs` class which will add the current model reference once instantiated
+  const Attrs = Ember.Object.extend(attrCPs, {
+    init() {
+      this._super(...arguments);
+      let model = this.get('_model');
+
+      validatableAttributes.forEach((attribute) => {
+        // Add a reference to the model in the deepest object
+        let path = attribute.split('.');
+        let lastObject = get(this, path.slice(0, path.length  - 1).join('.'));
+        if(isNone(get(lastObject, '_model'))) {
+          set(lastObject, '_model', model);
+        }
+      });
+    }
+  });
+
+  // Create `validations` class
+  return Ember.Object.extend(TopLevelProps, {
     model: null,
     attrs: null,
     isValidations: true,
 
+    validatableAttributes: computed.alias('_validatableAttributes').readOnly(),
+
     // Caches
     _validators: null,
     _debouncedValidations: null,
 
     // Private
-    _validationRules: null,
-    _validatableAttributes: null,
-    _parentValidations: null,
+    _validationRules: computed(function() {
+      return validationRules;
+    }).readOnly(),
+
+    _validatableAttributes: computed(function() {
+      return validatableAttributes;
+    }).readOnly(),
 
     validate,
     validateSync,
 
     init() {
       this._super(...arguments);
-      let validationRules = validations;
-      let parentValidations = this.get('_parentValidations');
-      let attrs = {};
-
-      if(parentValidations) {
-        validationRules = merge(merge({}, parentValidations.get('_validationRules')), validationRules);
-      }
-
-      let validatableAttributes = Object.keys(validationRules);
-      validatableAttributes.forEach((attribute) => {
-        attrs[attribute] = createCPValidationFor(attribute, validationRules[attribute]);
-      });
-
       this.setProperties({
-        _validatableAttributes: validatableAttributes,
-        _validationRules: validationRules,
+        attrs: Attrs.create({ _model: this.get('model') }),
         _validators: {},
-        _debouncedValidations: {},
-        attrs: Ember.Object.extend(attrs).create({
-          _model: this.get('model')
-        })
+        _debouncedValidations: {}
       });
-
-      createGlobalValidationProps(this);
     },
 
     destroy() {
       this._super(...arguments);
+      let validatableAttrs = get(this, 'validatableAttributes');
       let debouncedValidations = get(this, `_debouncedValidations`);
 
       // Cancel all debounced timers
-      Object.keys(debouncedValidations).forEach(attr => {
-        let attrCache = debouncedValidations[attr];
-        // Itterate over each attribute and cancel all of its debounced validations
-        Object.keys(attrCache).forEach(v => run.cancel(attrCache[v]));
+      validatableAttrs.forEach(attr => {
+        let attrCache = get(debouncedValidations, attr);
```

Comment:
i think the attr here will always be a simple property access, as debounceValidations is just a pojo. We should likely avoid the extra get here,just do

```
let attrCache = debounceValidations[attr] ;
if (attrCache === undefined) { return; }
Object.keys(attrCache).forEach( ... )
```